File: TurnMatrixM.m
function TM = TurnMatrixM(M, Rotation)
% Turn the Mueller matrix of an element M, an angle Rotation
%
%   VARIABLES:  M: Mueller Matrix of the element that we want to rotate
%               Rotation: Rotation angle
%   
%   USE:        TM = TurnMatrixM(M, Rotation)

TM = Mueller('rotated', -Rotation, 0.0) * M * Mueller('rotated', Rotation, 0.0);

File: TurnMatrixM.py
import numpy as np
from Mueller import Mueller

def TurnMatrixM(M, Rotation):
    """
    Rotates a Mueller matrix using Stokes-space rotation.

    Parameters
    ----------
    M : np.ndarray (4x4)
        Mueller matrix of the optical element
    Rotation : float
        Rotation angle (radians)

    Returns
    -------
    TM : np.ndarray (4x4)
        Rotated Mueller matrix
    """

    R_left = Mueller('rotated', -Rotation, 0.0)
    R_right = Mueller('rotated', Rotation, 0.0)

    TM = R_left @ M @ R_right

    return TM